src/prefect/infrastructure/docker.py

```diff
@@ -368,7 +372,7 @@ def _build_container_settings(
         docker_client: "DockerClient",
     ) -> Dict:
         network_mode = self._get_network_mode()
-        return dict(
+        return {**self.extra_kwargs or dict(), **dict(
```

Comment:
I think we'll want `extra_kwargs` to take precedence — that will allow users to override our inference behavior if they really want to. I think we should warn if the keys collide though, so I'd do something roughly like:

```
# Construct our dict first
options = dict(...)

extra_kwargs = self.extra_kwargs or {}
duplicate_keys = set(options.keys()).intersection(extra_kwargs.keys())
if duplicate_keys:
   # Throw a warning, use `list_repr` to format keys that overlap, note which will take precedence
    warnings.warn(...)
    
options.update(extra_kwargs)
return options
```